### PR TITLE
[Merged by Bors] - Remove strict fee recipient

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

Resolves: #3550

Remove the `--strict-fee-recipient` flag. It will cause missed proposals prior to the bellatrix transition.

### CODE REVIEW

**Review by pawanjay176 — APPROVED** [review]

File: validator_client/src/cli.rs [hunk 1]
```diff
@@ -269,19 +269,6 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
                     execution payload construction during proposals.")
                 .takes_value(false),
         )
-        .arg(
-            Arg::with_name("strict-fee-recipient")
```
Comment (michaelsproul) [inline validator_client/src/cli.rs:273]: Thoughts on keeping this flag as a no-op with a warning? Or if we're doing a v3.2.0 release we can just mark it backwards incompatible and continue with deletion

Comment (realbigsean) [reply validator_client/src/cli.rs:273]: That's probably more sensible

**Review by michaelsproul — APPROVED** [review]
Nice! I think we're good to merge this for v3.1.1 or v3.2.0 (whichever we decide on)

**Pull request closed without merging**